src/software/ai/motion_constraint/motion_constraint_visitor.cpp

```diff
@@ -87,6 +87,21 @@ void MotionConstraintVisitor::visit(const MoveGoalieToGoalLineTactic &tactic)
     current_motion_constraints.erase(TbotsProto::MotionConstraint::FRIENDLY_DEFENSE_AREA);
 }
 
+void MotionConstraintVisitor::visit(const PlaceBallTactic &tactic)
+{
+    current_motion_constraints.clear();
+}
+
+void MotionConstraintVisitor::visit(const PlaceBallMoveTactic &tactic)
+{
+    current_motion_constraints.clear();
+}
+
+void MotionConstraintVisitor::visit(const WallKickoffTactic &tactic)
+{
+    current_motion_constraints.clear();
+}
```

Comment:
We should always keep the `TbotsProto::MotionConstraint::FRIENDLY_GOAL` obstacle, since that's an actual physical object that we need to avoid. Either you would need to add it back here, or probably a better long term solution is to update the `buildMotionConstraintSet` function to add the motion constraint after calling this visiter! 
then and remove it from here as well:
https://github.com/UBC-Thunderbots/Software/blob/f18e01fa5f818d20b91a7c1a413f864cef7fdb1e/src/software/ai/motion_constraint/motion_constraint_set_builder.cpp#L27

Reply:
does FRIENDLY_GOAL encompass the inside of the net?

In last year's ball placement challenge, this motion constraint would have caused an issue with placing the ball that originated inside the net

Reply:
`FRIENDLY_GOAL` outlines the goals walls. So the robot can technically go into the net, but it may possibly not be able to touch the wall depending on where we put the destination and if it's within the obstacle, since the destination will be moved outside of the obstacle too far from the ball. Though in either case, we should definitely have the goal as an obstacle. I rather possibly not be able to get the ball in some edge case where it's in the corner of the net than try to drive through the goal's walls.